File: content/problems/2905-find-indices-with-index-and-value-difference-ii/solutions/eligible-prefix-minimum-and-maximum/Solution.java
class Solution {
    public int[] findIndices(int[] nums,int indexDifference,int valueDifference){
        int minValue=Integer.MAX_VALUE,maxValue=Integer.MIN_VALUE,minIndex=-1,maxIndex=-1;
        for(int j=indexDifference;j<nums.length;j++){
            int candidate=j-indexDifference;
            if(nums[candidate]<minValue){
                minValue=nums[candidate];
                minIndex=candidate;
            }
            if(nums[candidate]>maxValue){
                maxValue=nums[candidate];
                maxIndex=candidate;
            }
            if(nums[j]-minValue>=valueDifference)return new int[]{
                minIndex,j
            };
            if(maxValue-nums[j]>=valueDifference)return new int[]{
                maxIndex,j
            };
        }
        return new int[]{
            -1,-1
        };
    }
}

File: content/problems/2905-find-indices-with-index-and-value-difference-ii/solutions/eligible-prefix-minimum-and-maximum/Solution.kt
class Solution {
    fun findIndices(nums:IntArray,indexDifference:Int,valueDifference:Int):IntArray{
        var minValue=Int.MAX_VALUE;
        var maxValue=Int.MIN_VALUE;
        var minIndex=-1;
        var maxIndex=-1;
        for(j in indexDifference until nums.size){
            val candidate=j-indexDifference;
            if(nums[candidate]<minValue){
                minValue=nums[candidate];
                minIndex=candidate
            };
            if(nums[candidate]>maxValue){
                maxValue=nums[candidate];
                maxIndex=candidate
            };
            if(nums[j]-minValue>=valueDifference)return intArrayOf(minIndex,j);
            if(maxValue-nums[j]>=valueDifference)return intArrayOf(maxIndex,j)
        };
        return intArrayOf(-1,-1)
    }
}
